Purchases/Networking/Operations/GetOfferingsOperation.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  GetOfferingsOperation.swift
+//
+//  Created by Joshua Liebowitz on 11/19/21.
+
+import Foundation
+
+class GetOfferingsOperation: NetworkOperation {
+
+    let httpClient: HTTPClient
+    let authHeaders: [String: String]
+    private let offeringsCallbackCache: CallbackCache<OfferingsCallback>
+
+    init(httpClient: HTTPClient,
+         authHeaders: [String: String],
+         offeringsCallbackCache: CallbackCache<OfferingsCallback>) {
+        self.httpClient = httpClient
+        self.authHeaders = authHeaders
+        self.offeringsCallbackCache = offeringsCallbackCache
+    }
+
+    func getOfferings(appUserID: String, completion: @escaping OfferingsResponseHandler) {
+        guard let appUserID = try? appUserID.escapedOrError() else {
+            completion(nil, ErrorUtils.missingAppUserIDError())
+            return
+        }
+
+        let path = "/subscribers/\(appUserID)/offerings"
+        let offeringsCallback = OfferingsCallback(key: path, callback: completion)
+        if self.offeringsCallbackCache.add(callback: offeringsCallback) == .addedToExistingInFlightList {
+            return
+        }
+
+        httpClient.performGETRequest(serially: true,
+                                     path: path,
+                                     headers: authHeaders) { [weak self] (statusCode, maybeResponse, maybeError) in
+            guard let self = self else {
```

Comment:
is there a reason we only sometimes do this check?

Reply:
I think this is one of the places where we were seeing some weird behavior, and we wanted to see if backend was deallocated. 
We probably should should think more about how we want to handle `self` in these blocks- how we want to retain them.

Reply:
It is actually proving to be problematic with the new operation subclass, without holding a strong reference, this code isn't completing.